### PR TITLE
python311Packages.nanobind: 1.9.2 -> 2.0.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/wjakob/nanobind/blob/v2.0.0/docs/changelog.rst

cc @phiadaarr 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 314119` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 314119` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 314119` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.islpy</li>
    <li>python311Packages.islpy.dist</li>
    <li>python311Packages.loopy</li>
    <li>python311Packages.loopy.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.nanobind</li>
    <li>python311Packages.nanobind.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 314119` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.islpy</li>
    <li>python311Packages.islpy.dist</li>
    <li>python311Packages.loopy</li>
    <li>python311Packages.loopy.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.nanobind</li>
    <li>python311Packages.nanobind.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/nanobind/default.nix [hunk 1]
```diff
@@ -18,25 +18,26 @@
...
+    repo = "nanobind";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-yDFrhSIWywWw7ri5aHRPigi9ujDeazpJa4AVrkLx5BI=";
     fetchSubmodules = true;
   };
 
   disabled = pythonOlder "3.8";
 
-  nativeBuildInputs = [
+  build-system = [
```
Comment (phiadaarr) [inline pkgs/development/python-modules/nanobind/default.nix:34]: Thanks a lot for your cleanup and update! Could quickly explain to me why you did this change and what it implies? Many thanks :)

Comment (fabaff) [reply pkgs/development/python-modules/nanobind/default.nix:34]: The naming in the "pythonPackages space" was aligned with PEP-517. `build-system` should now contains the actual build system, e. g., `setuptools`, `hatchling`, `poetry-core`, etc.. `nativeBuildInputs`still works, same goes for `propagatedBuildInputs` which was replaced by `dependencies`.

**Comment by GaetanLepage** [comment]
Before merging this, we should fix the `islpy` failure. This is a regression.

**Comment by phiadaarr** [comment]
`islpy` build failure is tracked here: https://github.com/inducer/islpy/issues/134
`loopy` depends on `islpy`.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 314119` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 314119` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 314119` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.islpy</li>
    <li>python311Packages.islpy.dist</li>
    <li>python311Packages.loopy</li>
    <li>python311Packages.loopy.dist</li>
    <li>python311Packages.nanobind</li>
    <li>python311Packages.nanobind.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 314119` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.islpy</li>
    <li>python311Packages.islpy.dist</li>
    <li>python311Packages.loopy</li>
    <li>python311Packages.loopy.dist</li>
    <li>python311Packages.nanobind</li>
    <li>python311Packages.nanobind.dist</li>
  </ul>
</details>

**Review by phiadaarr — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
Thank you!

**Pull request merged**